src/v/kafka/server/tests/produce_consume_test.cc

```diff
@@ -305,13 +353,85 @@ FIXTURE_TEST(test_node_throughput_limits, prod_consume_fixture) {
             kafka_out_data_len += kafka_data_len;
         }
         const auto stop = ch::steady_clock::now();
+        const auto rate_estimated = rate_limit_out
+                                    - _rate_minimum * (ss::smp::count - 1);
         const auto time_estimated = ch::milliseconds(
-          total_size * 1000 / pershard_rate_limit_out);
+          total_size * 1000 / rate_estimated);
         BOOST_TEST_CHECK(
-          abs(stop - start - time_estimated) < time_estimated / 25,
-          "stop-start[" << stop - start << "] == time_estimated["
-                        << time_estimated << "] ±4%");
+          abs(stop - start - time_estimated)
+            < (time_estimated * tolerance_percent / 100),
+          "Egress time: stop-start["
+            << stop - start << "] ≈ time_estimated[" << time_estimated << "] ±"
+            << tolerance_percent
+            << "%"
+               " "
+            << abs(stop - start - time_estimated) << " "
+            << (time_estimated * tolerance_percent / 100));
+        return kafka_out_data_len;
     }
+};
+
+FIXTURE_TEST(test_node_throughput_limits_static, throughput_limits_fixure) {
+    // configure
+    constexpr int64_t pershard_rate_limit_in = 9_KiB;
+    constexpr int64_t pershard_rate_limit_out = 7_KiB;
+    constexpr size_t batch_size = 256;
+    config_set(
+      "kafka_throughput_limit_node_in_bps",
+      std::make_optional(pershard_rate_limit_in * ss::smp::count));
+    config_set(
+      "kafka_throughput_limit_node_out_bps",
+      std::make_optional(pershard_rate_limit_out * ss::smp::count));
+    config_set("fetch_max_bytes", batch_size);
+    config_set("max_kafka_throttle_delay_ms", 60'000ms);
+    config_set_window_width(200ms);
+    config_set_balancer_period(0ms);
+    config_set_rate_minimum(0);
+
+    wait_for_controller_leadership().get();
+    start();
+
+    // PRODUCE 10 KiB in smaller batches, check throttle but do not honour it,
+    // check that has to take 1 s
+    const size_t kafka_in_data_len = test_ingress(
+      pershard_rate_limit_in, batch_size, 3 /*%*/);
+
+    // CONSUME
+    const size_t kafka_out_data_len = test_egress(
+      kafka_in_data_len, pershard_rate_limit_out, batch_size, 3 /*%*/);
+
+    // otherwise test is not valid:
+    BOOST_REQUIRE_GT(kafka_in_data_len, kafka_out_data_len);
+}
+
+FIXTURE_TEST(test_node_throughput_limits_balanced, throughput_limits_fixure) {
```

Comment:
have you thought about ducktape tests? ultimately that is what we are going to need.

Reply:
Definitely, I'm just not ready to PR them yet. The UTs are basically the bare minimum.